pkg/webreg/webreg.go

```diff
@@ -401,6 +401,112 @@ const templateDefinitions = `
+</p>
+
+<h3 id="bundles"><a href="#bundles">Building Operator Bundles</a></h3>
+
+<p>
+Configuring <code>ci-operator</code> to build operator bundles from a repository
+is as simple as adding new entries to the <code>images</code> stanza, specifying
+which directory in the repository contains the operator manifests and what sorts
+of container image pull specification substitutions are necessary during bundle
+build time. Substitutions allow for the operator manifests to refer to images that
+were built from the repository during the test. The following example builds an
+operator, a bundle and replaces the operator's pull specification in manifests
+with the version built during the test:
+</p>
+
+<code>ci-operator</code> configuration:
+{{ yamlSyntax (index . "optionalOperatorBundleConfig") }}
+
+<h3 id="index"><a href="#index">Building an Index</a></h3>
+
+<p>
+If <code>ci-operator</code> is configured to build operator bundles from a
+repository, an index image will automatically be built to package those bundles.
+</p>
+
+<h3 id="tests"><a href="#tests">Running Tests</a></h3>
+
+<p>
+Once <code>ci-operator</code> builds the operator bundle and index, they are
+available to be used as a <code>CatalogSource</code> by OLM for deploying and
+testing the operator. Today, the only style of testing that's supported is an
+end-to-end test, implemented with the <a href="/workflow/optional-operators-cvp-common-aws"><code>optional-operators-cvp-common-aws</code></a>
+workflow. This workflow takes the following steps to set up the test environment:
+</p>
+
+<ul>
+  <li>deploy an ephemeral OpenShift cluster to test against</li>
+  <li>create a <code>Namespace</code> to install into</li>
+  <li>create a <code>OperatorGroup</code> and <code>CatalogSource</code> to configure OLM</li>
+  <li>create a <code>Subscription</code> for the operator under test</li>
+  <li>wait for the operator under test to install and deploy</li>
+</ul>
+
+<p>
+A user-provided test can expect to have <code>${KUBECONFIG}</code> set, with
+administrative privileges, and for the operator under test to be fully deployed
+at the time that the test begins. The following example runs a test in this manner:
+</p>
+
+<code>ci-operator</code> configuration:
+{{ yamlSyntax (index . "optionalOperatorTestConfig") }}
+`
+
+const optionalOperatorBundleConfig = `
+base_images:
+  ubi:               # imports the UBI base image for building
+    namespace: "ocp"
+    name: "ubi"
+    tag: "8"
+  operand:           # imports the latest operand image
+    namespace: "ocp"
+    name: "operand"
+    tag: "latest"
+images:
+- from: ubi
+  to: operator                         # the pipeline:operator image, built on top of ubi:8
+  dockerfile_path: Dockerfile
+- from: ubi
+  to: operator-bundle                  # the pipeline:bundle image, build on top of ubi:8
+  dockerfile_path: bundle.Dockerfile
+  operator_manifests: bundle/manifests # where operator manifests reside in the repository
```

Comment:
imho we should default this if unset to the operator sdk default (`bundle/manifests`)

Reply:
this looks like a pitfall to me though, I don't think I've ever seen a real operator repo that would have a simple path like this